test/e2e/annotations/affinity.go

```diff
@@ -32,11 +32,15 @@ import (
 	"k8s.io/ingress-nginx/test/e2e/framework"
 )
 
+// TODO(elvinefendi) merge this with Affinity tests in test/e2e/lua/dynamic_configuration.go
 var _ = framework.IngressNginxDescribe("Annotations - Affinity", func() {
 	f := framework.NewDefaultFramework("affinity")
 
 	BeforeEach(func() {
-		err := f.NewEchoDeploymentWithReplicas(2)
+		err := f.DisableDynamicConfiguration()
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
This is a known limitation of affinity implementation in dynamic mode as described at https://github.com/kubernetes/ingress-nginx/issues/2662. The core affinity features are supported in dynamic mode and there are tests for them at test/e2e/lua/dynamic_configuration.go. It is only https://github.com/kubernetes/ingress-nginx/pull/2794/files#diff-7d332a0e5ba59e06193678d5a83909d0R102 that's not supported yet as described in the issue.